blueocean-dashboard/src/main/js/components/RunDetailsPipeline.jsx

```diff
@@ -176,6 +176,31 @@ export class RunDetailsPipeline extends Component {
                 }
             case 'pipeline_end':
                 {
+                    // get all steps from the current run, we use the nodeKey and remove the last bit
+                    const keyArray = this.mergedConfig.nodeKey.split('/');
+                    keyArray.pop();
+                    if (keyArray[keyArray.length - 1] === 'nodes') {
+                        keyArray.pop();
+                    }
+                    const searchString = keyArray.join('/');
+                    // fire to remove all logs from cache
+                    removeLogs(searchString);
+                    /** we have to check now if all steps have been declared as finished,
+                     * if not we will remove them from the cache so we get the final information*/
+                    const notFinishedSteps = Object.keys(this.props.steps)
+                        .filter((item) => item.indexOf(searchString) !== -1)
```

Comment:
Mozilla examples refer to this method, I suppose I should try to adapt my brain: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/indexOf